pipelines/stripe_analytics/stripe_analytics.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from enum import Enum
+from typing import Any, Generator, Optional, Dict
+
+import dlt
+import stripe
+from dlt.extract.source import DltResource
+from dlt.pipeline import Pipeline
+from dlt.common import pendulum
+
+from .metrics import calculate_mrr, churn_rate
+
+
+class Endpoints(Enum):
+    customers: str = "Customer"
+    subscriptions: str = "Subscription"
+    prices: str = "Price"
+    discounts: str = "Coupon"
+    events: str = "Event"
+
+
+def stripe_get_data(
+    resource: Endpoints, start_date: Optional[Any] = None, end_date: Optional[Any] = None, **kwargs : Any
+) -> Dict[Any, Any]:
+    if start_date:
+        if isinstance(start_date, str):
+            start_date = datetime.strptime(start_date, "%Y-%m-%dT%H:%M:%SZ")
+        if isinstance(start_date, datetime):
+            # convert to unix timestamp
+            start_date = int(start_date.timestamp())
+    if end_date:
+        # convert to unix timestamp
+        end_date = int(end_date.timestamp())
+
+    if resource == Endpoints.subscriptions:
+        kwargs.update({"status": "all"})
+
+    resource_dict = getattr(stripe, resource.value).list(
+        created={"gte": start_date, "lt": end_date}, **kwargs
+    )
+    return dict(resource_dict)
+
+
+@dlt.source
+def stripe_source(
+    stripe_secret_key: str = dlt.secrets.value,
+    limit: int = 100,
+    get_all_data: bool = False,
```

Comment:
what is the use of `get_all_data`? is it useful to our users? on first look it stops loading after receiving first page of results. this you can do directly in `dlt` by doing `stripe_source().add_limit(1)`. https://dlthub.com/docs/general-usage/source#load-data-partially
IMO we do not need this option

Reply:
You suggest to load all data by default?
sure, I can use this `stripe_source().add_limit(1)` feature additionally